src/Form/Form.js

```diff
@@ -347,12 +381,17 @@ class Form {
             if (foundInputs.length < MAX_INPUTS_PER_FORM) {
                 foundInputs.forEach(input => this.addInput(input))
             } else {
-                if (shouldLog()) {
-                    console.log('The form has too many inputs, bailing.')
-                }
+                // This is rather extreme, but better safe than sorry
+                this.device.scanner.stopScanner('The form has too many inputs, bailing.')
+                return
```

Comment:
This is another crucial part. Previously, the `else` block would not return nor stop the scanner, thus a lot of work would keep happening after hitting the failsafes. TBH, stopping the scanner completely is probably overkill, but we've been burned before so I'm happy to go nuclear here.